walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,92 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;  
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    ArrayList<TimeRange> possibleTimes;
+
+    public FindMeetingQuery() {
+        possibleTimes = new ArrayList<TimeRange>();
+    }
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        long reqDuration = request.getDuration();
+        Collection<String> reqAttendees = request.getAttendees();
+
+        // Special cases
+        if (reqDuration > TimeRange.WHOLE_DAY.duration()) {
+            return possibleTimes;
+        }
+
+        // Collection of clashing events' (events where at least one attendee belongs to reqAttendees) timeranges
+        List<TimeRange> clashingEventTimes = new ArrayList<TimeRange>();
+
+        // Check each event in events collection
+        clashingEventTimes = events.stream()
+            .filter(event -> isClashingEvent(event, reqAttendees))
+            .map(event -> event.getWhen())
+            .collect(Collectors.toList()); 
+
+        // Check for case where there is no clashing event
+        int numEvents = clashingEventTimes.size();
+        if (numEvents == 0) {
```

Comment:
I would combine this with the line above, no need to pull out a separate variable unless there's more than two chained method calls or it doesn't fit on one line